fileshare/event_manager.go

```diff
@@ -301,148 +252,156 @@ func (em *EventManager) handleTransferFinishedEvent(eventJSON json.RawMessage) {
 		close(progressCh)
-		delete(em.transferSubscriptions, transfer.Id)
+		delete(em.transferSubscriptions, transfer.ID)
 	}
-	transfer.Finalized = true
-}
-
-// NewOutgoingTransfer used when we are sending files, because libdrop only emits the event a bit later
-func (em *EventManager) NewOutgoingTransfer(id, peer, path string) {
-	em.mutex.Lock()
-	defer em.mutex.Unlock()
 
-	if transfer, ok := em.transfers[id]; ok {
-		transfer.Peer = peer
-		transfer.Path = path
-	} else {
-		em.transfers[id] = NewOutgoingTransfer(id, peer, path)
-	}
+	delete(em.liveTransfers, transfer.ID)
 }
 
 // GetTransfers is used for listing transfers.
 // Returned transfers are sorted by date created from oldest to newest.
-// Returns copies of transfers, modifying them will not change EventManager state.
-func (em *EventManager) GetTransfers() []*pb.Transfer {
+func (em *EventManager) GetTransfers() ([]*pb.Transfer, error) {
 	em.mutex.Lock()
 	defer em.mutex.Unlock()
 
-	transfers := make([]*pb.Transfer, 0, len(em.transfers))
-	for _, transfer := range em.transfers {
-		transferCloned, ok := proto.Clone(transfer).(*pb.Transfer)
-		if !ok {
-			log.Printf("failed to cast cloned transfer %s", transfer.Id)
-			continue
-		}
-		transfers = append(transfers, transferCloned)
+	storageTransfers, err := em.storage.Load()
+	if err != nil {
+		return nil, fmt.Errorf("loading transfers from storage: %s", err)
+	}
+
+	transfers := make([]*pb.Transfer, 0, len(storageTransfers))
+	for _, storageTransfer := range storageTransfers {
+		updateTransferWithLiveData(storageTransfer, em.liveTransfers)
+		transfers = append(transfers, storageTransfer)
 	}
 
 	sort.Slice(transfers, func(i int, j int) bool {
 		return transfers[i].Created.AsTime().Before(transfers[j].Created.AsTime())
 	})
 
-	return transfers
+	return transfers, nil
 }
 
 // GetTransfer by ID.
-// Returns a copy, modifying it will not change EventManager state.
 func (em *EventManager) GetTransfer(transferID string) (*pb.Transfer, error) {
 	em.mutex.Lock()
 	defer em.mutex.Unlock()
+	return em.getTransfer(transferID)
+}
+func (em *EventManager) getTransfer(transferID string) (*pb.Transfer, error) {
+	transfer, err := getTransferFromStorage(transferID, em.storage)
+	if err != nil {
+		return nil, err
+	}
+	updateTransferWithLiveData(transfer, em.liveTransfers)
+	return transfer, nil
+}
 
-	transfer, ok := em.transfers[transferID]
+func getTransferFromStorage(id string, storage Storage) (*pb.Transfer, error) {
+	storageTransfers, err := storage.Load()
+	if err != nil {
+		return nil, fmt.Errorf("loading transfers from storage: %s", err)
+	}
+	storageTransfer, ok := storageTransfers[id]
 	if !ok {
 		return nil, ErrTransferNotFound
 	}
+	return storageTransfer, nil
+}
 
-	transferCloned, ok := proto.Clone(transfer).(*pb.Transfer)
+// Storage doesn't contain momentary info about transfer progress, so update it from liveTransfers
+func updateTransferWithLiveData(transfer *pb.Transfer, liveTransfers map[string]*LiveTransfer) {
```

Comment:
I'd change this function to return `*pb.Transfer`. It's not necessary  but assignment is more verbose/explicit.